modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -468,7 +474,19 @@ private List<CompletableFuture<AsyncSqlCursor<List<Object>>>> query0(
 
         CompletableFuture<Void> start = new CompletableFuture<>();
 
+        boolean needStartTx = false;
+
+        for (SqlNode sqlNode : nodes) {
+           if (SqlKind.DML.contains(sqlNode.getKind()) || SqlKind.QUERY.contains(sqlNode.getKind())) {
```

Comment:
why you need transaction for SELECT statement?

Reply:
Honestly, I don`t know how select will be processed without tx.

Reply:
ok, which type of query we could have here, outside of DML and QUERY?

Reply:
i want to filter DDL here